### PR TITLE
Turn off GFN-FF fragmentation if it is not needed.

### PR DESCRIPTION
**Pull request by MtoLStoN**
Signed-off-by: MtoLStoN <70513124+MtoLStoN@users.noreply.github.com>

### CODE REVIEW

File: src/gfnff/gfnff_ini.f90 [hunk 1]
```diff
@@ -1833,12 +1833,18 @@ integer function itabrow6(i)
 ! all done
 
       topo%maxsystem = 5000
-      !if(mol%n.gt.500)then
-      call fragmentize(mol%n,mol%at,mol%xyz,topo%maxsystem,500,rab,topo%nb, &
-         & topo%ispinsyst,topo%nspinsyst,topo%nsystem)
-      !else
-      !   nsystem=1
-      !endif
+      !if(mol%n.gt.800)then
```
Comment (awvwgk) [inline src/gfnff/gfnff_ini.f90:1836]:
```suggestion
```

**Review by awvwgk — APPROVED** [review]
Thanks for sharing. Looks good to me.

**Comment by MtoLStoN** [comment]
Should now also raise less errors with the default hessian.

File: src/gfnff/frag_hess.f90 [hunk 2]
```diff
@@ -303,6 +306,11 @@ subroutine fragmentize(nspin, at, xyz, maxsystem, maxmagnat, jab, neigh, ispinsy
            end do ! End Loop over  i from 1 to count(visited)
            ass = maxloc(nspinsyst, 1)
 
+            if (nsystem .ge. nspin) then
+               call env%warning("Created more Fragments than atoms. This does not look good. Turning off fragmentization.")
```
Comment (awvwgk) [inline src/gfnff/frag_hess.f90:310]:
```suggestion
               call env%warning("Created more fragments than atoms. This does not look good. Turning off fragmentation.")
```

**Pull request merged**